x/auth/types/codec.go

```diff
@@ -2,32 +2,31 @@ package types
 
 import (
 	"github.com/cosmos/cosmos-sdk/codec"
-	"github.com/cosmos/cosmos-sdk/x/auth/exported"
+	"github.com/cosmos/cosmos-sdk/codec/types"
 )
 
-// Codec defines the interface needed to serialize x/auth state. It must be
-// aware of all concrete account types.
-type Codec interface {
-	codec.Marshaler
-
-	MarshalAccount(acc exported.Account) ([]byte, error)
-	UnmarshalAccount(bz []byte) (exported.Account, error)
-
-	MarshalAccountJSON(acc exported.Account) ([]byte, error)
-	UnmarshalAccountJSON(bz []byte) (exported.Account, error)
-}
-
 // RegisterCodec registers the account interfaces and concrete types on the
 // provided Amino codec.
 func RegisterCodec(cdc *codec.Codec) {
-	cdc.RegisterInterface((*exported.ModuleAccountI)(nil), nil)
-	cdc.RegisterInterface((*exported.GenesisAccount)(nil), nil)
-	cdc.RegisterInterface((*exported.Account)(nil), nil)
-	cdc.RegisterConcrete(&BaseAccount{}, "cosmos-sdk/Account", nil)
+	cdc.RegisterInterface((*ModuleAccountI)(nil), nil)
+	cdc.RegisterInterface((*GenesisAccount)(nil), nil)
+	cdc.RegisterInterface((*AccountI)(nil), nil)
+	cdc.RegisterConcrete(&BaseAccount{}, "cosmos-sdk/BaseAccount", nil)
 	cdc.RegisterConcrete(&ModuleAccount{}, "cosmos-sdk/ModuleAccount", nil)
 	cdc.RegisterConcrete(StdTx{}, "cosmos-sdk/StdTx", nil)
 }
 
+// RegisterInterface associates protoName with AccountI interface
+// and creates a registry of it's concrete implementations
+func RegisterInterfaces(registry types.InterfaceRegistry) {
+	registry.RegisterInterface(
+		"cosmos_sdk.auth.v1.AccountI",
+		(*AccountI)(nil),
+		&BaseAccount{},
+		&ModuleAccount{},
+	)
+}
+
 // RegisterKeyTypeCodec registers an external concrete type defined in
 // another module for the internal ModuleCdc.
 func RegisterKeyTypeCodec(o interface{}, name string) {
```

Comment:
@fedekunze would you still need this for ethermint, even after removing auth/Codec